src/algebra/algebra/linear_algebra.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Yury Kudryashov
+-/
+import algebra.algebra.basic
+import linear_algebra.tensor_product
+
+/-!
+# Facts about algebras involving linear maps and tensor products
+
+We move a few basic statements about algebras out of `algebra.algebra.basic`,
+in order to avoid importing `linear_algebra.tensor_product` unnecessarily.
+-/
+
+universes u v w
+
+namespace algebra
+
+open_locale tensor_product
+open module
+
+section
+
+variables (R A : Type*) [comm_semiring R] [semiring A] [algebra R A]
+
+/-- The multiplication in an algebra is a bilinear map.
+
+A weaker version of this for semirings exists as `add_monoid_hom.mul`. -/
+def lmul : A →ₐ[R] (End R A) :=
+{ map_one' := by { ext a, exact one_mul a },
+  map_mul' := by { intros a b, ext c, exact mul_assoc a b c },
+  map_zero' := by { ext a, exact zero_mul a },
+  commutes' := by { intro r, ext a, dsimp, rw [smul_def] },
+  .. (show A →ₗ[R] A →ₗ[R] A, from linear_map.mk₂ R (*)
+  (λ x y z, add_mul x y z)
+  (λ c x y, by rw [smul_def, smul_def, mul_assoc _ x y])
+  (λ x y z, mul_add x y z)
+  (λ c x y, by rw [smul_def, smul_def, left_comm])) }
```

Comment:
I think this whole thing golfs to `algebra.lsmul`, but that's out of scope here.

Reply:
I don't see what you mean yet. Feel free to add, or tell me what to do, or leave it for later! :-)

Reply:
I'm suggesting that a defeq implementation is just:
```suggestion
lsmul R A
```
although this would have to then import `algebra.algebra.tower`, so I think is future work not for this PR.